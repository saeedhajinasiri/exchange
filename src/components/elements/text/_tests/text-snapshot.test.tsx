import React from "react";
import {cleanup, render} from "test";
import Text from '../text'

afterEach(cleanup)

describe('Test Text UI', () => {
  it('Default', () => {
    const {asFragment} = render(<Text color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with className prop', () => {
    const {asFragment} = render(<Text className={"test-class-name"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with style prop', () => {
    const {asFragment} = render(<Text style={{}} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Text Colors', () => {
  it('color=primary', () => {
    const {asFragment} = render(<Text color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=secondary', () => {
    const {asFragment} = render(<Text color={"secondary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=tertiary', () => {
    const {asFragment} = render(<Text color={'tertiary'}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=grey', () => {
    const {asFragment} = render(<Text color={'grey'}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=light', () => {
    const {asFragment} = render(<Text color={'light'}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=dark', () => {
    const {asFragment} = render(<Text color={'dark'}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Text Typographies', () => {
  it('typography=sm', () => {
    const {asFragment} = render(<Text typography={'sm'} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('typography=base', () => {
    const {asFragment} = render(<Text typography={'base'} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('typography=lg', () => {
    const {asFragment} = render(<Text typography={'lg'} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Text Types', () => {
  it('type=regular', () => {
    const {asFragment} = render(<Text type={"regular"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('type=bold', () => {
    const {asFragment} = render(<Text type={"bold"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('type=light', () => {
    const {asFragment} = render(<Text type={"light"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('type=medium', () => {
    const {asFragment} = render(<Text type={"medium"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Text Aligns', () => {
  it('align="center', () => {
    const {asFragment} = render(<Text align={"center"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('align="justify', () => {
    const {asFragment} = render(<Text align={"justify"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('align="left', () => {
    const {asFragment} = render(<Text align={"left"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('align="right', () => {
    const {asFragment} = render(<Text align={"right"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Text Variants', () => {
  it('variant=span', () => {
    const {asFragment} = render(<Text variant={"span"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('variant=h1', () => {
    const {asFragment} = render(<Text variant={"h1"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('variant=p', () => {
    const {asFragment} = render(<Text variant={"p"} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Text Disabled', () => {
  it('disabled=true', () => {
    const {asFragment} = render(<Text disabled={true} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('disabled=false', () => {
    const {asFragment} = render(<Text disabled={false} color={"primary"}>children</Text>);
    expect(asFragment()).toMatchSnapshot();
  });
});
