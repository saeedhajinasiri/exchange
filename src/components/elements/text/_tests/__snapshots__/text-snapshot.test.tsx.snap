// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Text Aligns align="center 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-center font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Aligns align="justify 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-justify font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Aligns align="left 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Aligns align="right 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-right font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Colors color=dark 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-grey-dark text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Colors color=grey 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Colors color=light 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-grey-light text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Colors color=primary 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Colors color=secondary 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-secondary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Colors color=tertiary 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-tertiary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Disabled disabled=false 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Disabled disabled=true 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Types type=bold 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-bold"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Types type=light 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-light"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Types type=medium 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-medium"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Types type=regular 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Typographies typography=base 1`] = `
<DocumentFragment>
  <p
    class="text-m-base sm:text-t-base md:text-d-base text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Typographies typography=lg 1`] = `
<DocumentFragment>
  <p
    class="text-m-lg sm:text-t-lg md:text-d-lg text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Typographies typography=sm 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text UI Default 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text UI Test with className prop 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular test-class-name"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text UI Test with style prop 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Variants variant=h1 1`] = `
<DocumentFragment>
  <h1
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </h1>
</DocumentFragment>
`;

exports[`Test Text Variants variant=p 1`] = `
<DocumentFragment>
  <p
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </p>
</DocumentFragment>
`;

exports[`Test Text Variants variant=span 1`] = `
<DocumentFragment>
  <span
    class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
  >
    children
  </span>
</DocumentFragment>
`;
